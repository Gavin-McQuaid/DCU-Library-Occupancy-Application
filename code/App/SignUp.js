import React, {Component} from 'react';
import {View, Text, Alert, TextInput, StyleSheet, StatusBar, Image,ScrollView,KeyboardAvoidingView, TouchableOpacity,TouchableHighlight} from 'react-native';
import {StackNavigator} from 'react-navigation';
import {hook} from 'cavy';
class SignUp extends Component{

	static navigationOptions = {
		title: "SignUp"
	}
	constructor(props){
		super(props)
		this.state = {
			NewUserEmail: '',
			NewUserPassword: '',
			NewUserPasswordConfirmation:'',
			EmailError: '',
			PasswordError: '',
			PasswordConfirmationError: ''
		
		}
	}
	// Validates email account entered to check it's a DCU email address
	ValidateEmail = () => {
		var emailregex = /^[a-zA-Z0-9_.+-]+@(?:(?:[a-zA-Z0-9-]+\.)?[a-zA-Z]+\.)?(dcu)\.ie$/;
		if(!emailregex.test(String(this.state.NewUserEmail).toLowerCase())){
			this.setState({EmailError: "Valid DCU email required"})
			return false
		}
		else{
			this.setState({EmailError: ""})
			return true
		}
	}

	// Validates password entered to check it's at least eight characters
	ValidatePassword = () => {	
	    if(this.state.NewUserPassword.length < 8){
			this.setState({PasswordError: "Passwords must not be empty and at least 8 characters long"})
			return false
		}
		else{
			this.setState({PasswordError: ""})
			return true
		}
	}

	// Checks password entered matches confirmation field
	PasswordsMatch = () => {
		if(this.state.NewUserPassword !== this.state.NewUserPasswordConfirmation || this.state.NewUserPasswordConfirmation.length < 8){
			this.setState({PasswordConfirmationError: "Passwords don't match"})
			return false
		}
		else{
			this.setState({PasswordConfirmationError: ""})
			return true
		}
	}

	// One final validation check triggered when Create Account button is pressed
	FinalValidation = () => {
		emailOk = this.ValidateEmail();
		passwordOk = this.ValidatePassword();
		passwordsMatch = this.PasswordsMatch();
		if(emailOk  && passwordOk && passwordsMatch){
			this.CreateAccount();
		}
		else{
		Alert.alert("Your details didn't meet validation requirements")
		}
	}

	// Function that creates a new user account
	CreateAccount = () => {	
		const {NewUserEmail} = this.state;
		const {NewUserPassword} = this.state;
		const {NewUserPasswordConfirmation} = this.state;
		fetch('http://ec2-34-244-240-250.eu-west-1.compute.amazonaws.com/CreateAccount.php', {
			method: 'POST',
			headers: {
				'Accept' : 'application/json',
				'Content-Type' : 'applcation/json'
			},
			body: JSON.stringify({
				email: NewUserEmail,
				password: NewUserPassword,
				password_confirmation: NewUserPasswordConfirmation

			}) 
		}).then((response) => response.json())
      .then((responseJson) => {
 
 		
		if(responseJson == "User registered successfully"){
			this.GenerateCode(NewUserEmail)
			this.props.navigation.navigate("Login", {screen: "Login"})
		}
		else{
			Alert.alert("Email already in use! Please use another email account")
		}
      }).catch((error) => {
        console.error(error);
      })
      ;}

      GenerateCode = (email) => {
		fetch('http://ec2-34-244-240-250.eu-west-1.compute.amazonaws.com/GenerateCode.php', {
			method: 'POST',
			headers: {
				'Accept' : 'application/json',
				'Content-Type' : 'applcation/json'
			},
			body: JSON.stringify({
				email: email
			}) 
		}).then((response) => response.json())
      .then((responseJson) => {

		if(responseJson == "Success"){
			Alert.alert("Account created successfully, check your email for authentication code");
		}
		else{
			Alert.alert("Something went wrong");
		}

      }).catch((error) => {
        console.error(error);
      });

	}
	
	render() {
		return (
		<ScrollView behavior = 'padding' style = {styles.container}>
			<View style = {styles.logoContainer}>
				<Text style = {styles.title}> DCU Library App 
				</Text>
				<Image
					style = {styles.logo}
					source = {require('./images/logo.png')} 
					/>		
			</View>
			<View style = {styles.formContainer}>
				<StatusBar
					barStyle = 'light-content'
				/>	
				<TextInput
					ref = {this.props.generateTestHook('SignUp.EmailTextInput')}
					placeholder = "Email"
					placeholderTextColor = 'white'
					returnKeyType = 'next'
					autoCapitalize = 'none'
					KeyboardType = 'email-address'
					autoCorrect = {false}
					onChangeText ={NewUserEmail => this.setState({NewUserEmail},() => this.ValidateEmail())}
					style = {styles.input}
				/>
				<Text style = {styles.errorText} >{this.state.EmailError}
				</Text>
				<TextInput
					ref = {this.props.generateTestHook('SignUp.PasswordTextInput')}
					placeholder = "Password"
					placeholderTextColor = 'white'
					returnKeyType = 'next'
					autoCapitalize = 'none'
					secureTextEntry
					style = {styles.input}
					onChangeText ={NewUserPassword => this.setState({NewUserPassword}, () => this.ValidatePassword())}
				/>
				<Text style = {styles.errorText} >{this.state.PasswordError}
				</Text>
				<TextInput
					ref = {this.props.generateTestHook('SignUp.ConfirmPasswordTextInput')}
					placeholder = "Confirm Password"
					placeholderTextColor = 'white'
					returnKeyType = 'go'
					autoCapitalize = 'none'
					secureTextEntry
					style = {styles.input}
					onChangeText ={NewUserPasswordConfirmation => this.setState({NewUserPasswordConfirmation},() => this.PasswordsMatch() )}
				/>
				<Text style = {styles.errorText} >{this.state.PasswordConfirmationError}
				</Text>
				<TouchableOpacity 
					ref = {this.props.generateTestHook('SignUp.CreateAccountButton')}
					style = {styles.buttonContainer}
					onPress={() => this.FinalValidation()}>
					<Text style = {styles.buttontext}> Create Account 
					</Text>
				</TouchableOpacity>
				<View style = {styles.signUpTextCont}>
					<Text style = {styles.signUpText}> Already have an account? 
					</Text>
					<TouchableHighlight
						ref = {this.props.generateTestHook('SignUp.SignInButton')}
						onPress={() => this.props.navigation.navigate("Login", {screen: "Login"})}>
						<Text style = {styles.signUpButton}> Sign in 
						</Text>
					</TouchableHighlight>
				</View>
			</View>
		</ScrollView>
		)
	}
}
export default hook(SignUp);
const styles = StyleSheet.create({
  container: {
	backgroundColor : 'cornflowerblue',  
    flex: 1,
	padding : 20
  },

  logoContainer: {
	alignItems : 'center',
	flexGrow : 1,
	justifyContent : 'center'
  },

  logo: {
	width : 200,
	height : 150
  },

  title: {
	color : 'white',
	marginBottom : 10,
	fontSize : 30,
	opacity : 0.9,
	justifyContent: 'center'
	
  },

  signUpTextCont: {
	flexGrow : 1,
	alignItems : 'flex-end',
	justifyContent : 'center',
	paddingVertical : 16,
	flexDirection : 'row'
  },

  signUpText: {
	color : 'rgba(255,255,255,.5)',
	fontSize : 16  
  },

  signUpButton: {
	color : 'white',
	fontWeight : '500',
	fontSize : 16  
  },

  errorText:{
  	color: 'white'
  },

  buttonContainer : {
	backgroundColor :'steelblue',
	paddingVertical : 15,
	marginBottom : 15,
	borderRadius : 25
   },

  buttontext : {
	color : 'white',
	textAlign : 'center',
	fontWeight : '400'
  },
  
  input: { 
	height: 40,
	backgroundColor : 'rgba(255,255,255,0.2)',
	marginBottom : 15,
	color : 'white',
	paddingHorizontal : 10,
	borderRadius : 25
	}
});