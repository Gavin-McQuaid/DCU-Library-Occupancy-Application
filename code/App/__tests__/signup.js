import React from 'react';
import SignUp from '../SignUp';
import renderer from 'react-test-renderer';
import {shallow, mount,render,configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import mockStore from 'redux-mock-store';
const Text = require('Text');
const TouchableHighlight = require('TouchableHighlight');
const ReactTestRenderer = require('react-test-renderer');
const TouchableOpacity = require('TouchableOpacity');
const TextInput = require('TextInput');
const Image = require('Image');

var ReactTestUtils = require('react-dom/test-utils')
configure({ adapter: new Adapter() })

describe('SignUp', () =>{
	it('text shows', () =>{
		const instance0 = ReactTestRenderer.create(
			<Text style = {{}}> DCU Library App </Text>	
		);
		});
	it('image shows', () =>{
		const instance1 = ReactTestRenderer.create(
			<Image style = {{}} source = {require('./Images/logo.png')}> </Image>
		);
		
	});
	it('TextInput 1', () => {
		const instance2 = ReactTestRenderer.create(
		<TextInput placeholder = 'Email'></TextInput>
		);	
	});
	it('TextInput 2', () => {
		const instance2 = ReactTestRenderer.create(
		<TextInput placeholder = 'Password'></TextInput>
		);	
	});
	it('TextInput 3', () => {
		const instance2 = ReactTestRenderer.create(
		<TextInput placeholder = 'Confirm Password'></TextInput>
		);	
	});
	it('touchable opacity 1 shows', () =>{
		const instance1 = ReactTestRenderer.create(
			<TouchableOpacity style ={{}}>
			<Text> Create Account</Text>
			</TouchableOpacity>
		);
	});
	it('touchable highlight shows', () =>{
		const instance = ReactTestRenderer.create(
			<TouchableHighlight style = {{}}>
			<Text> Sign in </Text>
			</TouchableHighlight>
		);
	});
	it('touchHighlight touch works', () => {
		const wrapper = shallow(<SignUp />);
		const touchHighlight = wrapper.find('TouchableHighlight');
		const store = mockStore({});
		const navigation = {navigate : jest.fn()};
		expect(renderer.create(<SignUp store = {store} navigation = {navigation} />)).toMatchSnapshot();
	});
		it ('Text Inputs works', () => {
		const wrapper = shallow(<SignUp />);
		const TextInput1= wrapper.find('TextInput').at(0);
		const TextInput2= wrapper.find('TextInput').at(1);
		const TextInput3= wrapper.find('TextInput').at(2);
		const credentials = {username : 'ed@dcu.ie' , password : 'hahaha', confirmpassword : 'hahaha'};
		TextInput1.value = credentials.username;
		expect(TextInput1.value).toBe('ed@dcu.ie');
		TextInput2.value = credentials.password;
		expect(TextInput2.value).toBe('hahaha');
		TextInput3.value = credentials.confirmpassword;
		expect(TextInput3.value).toBe('hahaha');
	});
	it('touchableOpacity 1 touch works', () => {
		const wrapper = shallow(<SignUp />);
		const touchOpacity1 = wrapper.find('TouchableOpacity').at(0);
		const store = mockStore({});
		const navigation = {navigate : jest.fn()};
		expect(renderer.create(<SignUp store = {store} navigation = {navigation} />)).toMatchSnapshot();
	});
		it ('Check Sign Up works', () => {
		const wrapper = shallow(<SignUp />);
		const TextInput1= wrapper.find('TextInput').at(0);
		const TextInput2= wrapper.find('TextInput').at(1);
		const TextInput3= wrapper.find('TextInput').at(2);
		const credentials = {username : 'ed@dcu.ie' , password : 'hahaha', confirmpassword : 'hahaha'};
		TextInput1.value = credentials.username;
		expect(TextInput1.value).toBe('ed@dcu.ie');
		TextInput2.value = credentials.password;
		expect(TextInput2.value).toBe('hahaha');
		TextInput3.value = credentials.confirmpassword;
		expect(TextInput3.value).toBe('hahaha');
		const touchOpacity1 = wrapper.find('TouchableOpacity').at(0);
		const store = mockStore({});
		const navigation = {navigate : jest.fn()};
		expect(renderer.create(<SignUp store = {store} navigation = {navigation} />)).toMatchSnapshot();
	});
	
			const rendered = renderer.create(<SignUp/>).toJSON();
		expect(rendered).toMatchSnapshot();
	
	
});