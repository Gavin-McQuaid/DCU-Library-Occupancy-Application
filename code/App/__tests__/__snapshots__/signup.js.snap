// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<RCTScrollView
  behavior="padding"
  style={
    Object {
      "backgroundColor": "cornflowerblue",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 0.9,
          }
        }
      >
         DCU Library App
      </Text>
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 150,
            "width": 200,
          }
        }
      />
    </View>
    <View
      style={undefined}
    >
      <TextInput
        KeyboardType="email-address"
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="Email"
        placeholderTextColor="white"
        returnKeyType="next"
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.2)",
            "borderRadius": 25,
            "color": "white",
            "height": 40,
            "marginBottom": 15,
            "paddingHorizontal": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="white"
        returnKeyType="next"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.2)",
            "borderRadius": 25,
            "color": "white",
            "height": 40,
            "marginBottom": 15,
            "paddingHorizontal": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Confirm Password"
        placeholderTextColor="white"
        returnKeyType="go"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.2)",
            "borderRadius": 25,
            "color": "white",
            "height": 40,
            "marginBottom": 15,
            "paddingHorizontal": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "steelblue",
            "borderRadius": 25,
            "marginBottom": 15,
            "opacity": 1,
            "paddingVertical": 15,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontWeight": "400",
              "textAlign": "center",
            }
          }
        >
           Create Account
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "center",
            "paddingVertical": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(255,255,255,.5)",
              "fontSize": 16,
            }
          }
        >
           Already have an account?
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
             Sign in
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`SignUp Check Sign Up works 1`] = `
<RCTScrollView
  behavior="padding"
  style={
    Object {
      "backgroundColor": "cornflowerblue",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 0.9,
          }
        }
      >
         DCU Library App
      </Text>
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 150,
            "width": 200,
          }
        }
      />
    </View>
    <View
      style={undefined}
    >
      <TextInput
        KeyboardType="email-address"
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="Email"
        placeholderTextColor="white"
        returnKeyType="next"
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.2)",
            "borderRadius": 25,
            "color": "white",
            "height": 40,
            "marginBottom": 15,
            "paddingHorizontal": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="white"
        returnKeyType="next"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.2)",
            "borderRadius": 25,
            "color": "white",
            "height": 40,
            "marginBottom": 15,
            "paddingHorizontal": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Confirm Password"
        placeholderTextColor="white"
        returnKeyType="go"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.2)",
            "borderRadius": 25,
            "color": "white",
            "height": 40,
            "marginBottom": 15,
            "paddingHorizontal": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "steelblue",
            "borderRadius": 25,
            "marginBottom": 15,
            "opacity": 1,
            "paddingVertical": 15,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontWeight": "400",
              "textAlign": "center",
            }
          }
        >
           Create Account
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "center",
            "paddingVertical": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(255,255,255,.5)",
              "fontSize": 16,
            }
          }
        >
           Already have an account?
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
             Sign in
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`SignUp touchHighlight touch works 1`] = `
<RCTScrollView
  behavior="padding"
  style={
    Object {
      "backgroundColor": "cornflowerblue",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 0.9,
          }
        }
      >
         DCU Library App
      </Text>
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 150,
            "width": 200,
          }
        }
      />
    </View>
    <View
      style={undefined}
    >
      <TextInput
        KeyboardType="email-address"
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="Email"
        placeholderTextColor="white"
        returnKeyType="next"
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.2)",
            "borderRadius": 25,
            "color": "white",
            "height": 40,
            "marginBottom": 15,
            "paddingHorizontal": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="white"
        returnKeyType="next"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.2)",
            "borderRadius": 25,
            "color": "white",
            "height": 40,
            "marginBottom": 15,
            "paddingHorizontal": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Confirm Password"
        placeholderTextColor="white"
        returnKeyType="go"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.2)",
            "borderRadius": 25,
            "color": "white",
            "height": 40,
            "marginBottom": 15,
            "paddingHorizontal": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "steelblue",
            "borderRadius": 25,
            "marginBottom": 15,
            "opacity": 1,
            "paddingVertical": 15,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontWeight": "400",
              "textAlign": "center",
            }
          }
        >
           Create Account
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "center",
            "paddingVertical": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(255,255,255,.5)",
              "fontSize": 16,
            }
          }
        >
           Already have an account?
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
             Sign in
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`SignUp touchableOpacity 1 touch works 1`] = `
<RCTScrollView
  behavior="padding"
  style={
    Object {
      "backgroundColor": "cornflowerblue",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexGrow": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "justifyContent": "center",
            "marginBottom": 10,
            "opacity": 0.9,
          }
        }
      >
         DCU Library App
      </Text>
      <Image
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 150,
            "width": 200,
          }
        }
      />
    </View>
    <View
      style={undefined}
    >
      <TextInput
        KeyboardType="email-address"
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="Email"
        placeholderTextColor="white"
        returnKeyType="next"
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.2)",
            "borderRadius": 25,
            "color": "white",
            "height": 40,
            "marginBottom": 15,
            "paddingHorizontal": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="white"
        returnKeyType="next"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.2)",
            "borderRadius": 25,
            "color": "white",
            "height": 40,
            "marginBottom": 15,
            "paddingHorizontal": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        onChangeText={[Function]}
        placeholder="Confirm Password"
        placeholderTextColor="white"
        returnKeyType="go"
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.2)",
            "borderRadius": 25,
            "color": "white",
            "height": 40,
            "marginBottom": 15,
            "paddingHorizontal": 10,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "steelblue",
            "borderRadius": 25,
            "marginBottom": 15,
            "opacity": 1,
            "paddingVertical": 15,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontWeight": "400",
              "textAlign": "center",
            }
          }
        >
           Create Account
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "center",
            "paddingVertical": 16,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "rgba(255,255,255,.5)",
              "fontSize": 16,
            }
          }
        >
           Already have an account?
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "500",
              }
            }
          >
             Sign in
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
